packages/core/utils/lib/convert-query-params.js

```diff
@@ -218,7 +218,80 @@ const convertFieldsQueryParams = (fields, depth = 0) => {
   throw new Error('Invalid fields parameter. Expected a string or an array of strings');
 };
 
-const convertFiltersQueryParams = filters => filters;
+const convertFiltersQueryParams = (filters, schema) => {
+  // Filters need to be either an array or an object
+  // Here we're only checking for 'object' type since typeof [] => object and typeof {} => object
+  if (!isObject(filters)) {
+    throw new Error('The filters parameter must be an object or an array');
+  }
+
+  const sanitizeFilters = (filters, schema) => {
```

Comment:
it doesn't look like that function needs to be redefined on each convertFiltersQUeryParams call. You could either recurse on convertFilters or put the sanitize outside

Reply:
Good catch